Update

Sintraqos opened the pull request.

Sintraqos committed:
Update

gglerum commented on PortfolioProject/PortfolioProject.API/src/main/java/com/sintraqos/portfolioproject/api/review/GameReviewAPI.java:
```diff
@@ -0,0 +1,159 @@
+    }
+
+    private void generateGames() {
+        // Directly creating objects with random reviews and scores
```
het had denk ik mooier geweest als deze uit een bron bestand werden uitgelezen zoals een csv, json of xml.

gglerum commented on PortfolioProject/PortfolioProject.API/pom.xml:
Is het de bedoeling dat deze module gaat samenwerken met PorfolioProjectAPI?

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Core/pom.xml:
```diff
@@ -0,0 +1,64 @@
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <source>21</source>
```
dit is een andere versie dan je in je properties hebt staan: 21 ipv 22.

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Core/src/main/resources/application.properties:
je hebt thymeleaf configuratie opties, maar dit project maakt geen gebruik van Thymeleaf templates? Even dit bestand opschonen zodat alles wat niet wordt gebruikt of uit staat met commentaar weg is.

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Core/src/main/java/com/sintraqos/portfolioprojectAPI/CoreBeans.java:
```diff
@@ -0,0 +1,34 @@
+@Configuration
+public class CoreBeans {
+
+    // Display all current loaded beans 
```
Alle code dat uit staat met comments mag weg.

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Core/src/main/java/com/sintraqos/portfolioprojectAPI/Main.java:
```diff
@@ -0,0 +1,23 @@
+    /**
+     * After the initialization of Spring-Boot this will start
+     */
+    @EventListener(ApplicationReadyEvent.class)
```
lege event listener mag ook weg.

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Game/src/main/java/com/sintraqos/portfolioprojectAPI/game/service/GameService.java:
```diff
@@ -0,0 +1,95 @@
+    public List<GameDTO> getRecentlyAddedGames(){
+        List<GameDTO> gameList = new ArrayList<>();
+        // TODO: Get all new releases
+        for(GameEntity game : getGame.getGames("")){
```
misschien een specifieke methode voor toevoegen? het is nu onduidelijk wat getGame.getGames("") terug geeft. Als ik de use case bekijk dan verwacht ik eigenlijk geen resultaten terug.

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Game/src/main/java/com/sintraqos/portfolioprojectAPI/game/GameBeans.java:
waar is dit bestand voor?  hij lijkt hier een beetje tussen packages te zweven.

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Game/src/main/java/com/sintraqos/portfolioprojectAPI/game/DTO/GameDTO.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public GameDTO(GameEntity game) {
+        if (game == null) {
```
vanwaar kan een DTO gemaakt worden van een game die niet bestaat?

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.Shared/pom.xml:
```diff
@@ -0,0 +1,20 @@
+
+    <artifactId>PortfolioProjectAPI.Shared</artifactId>
+
+    <properties>
```
deze properties zou je niet opnieuw hoeven te definiëren als ze al in de parent pom staan.

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.WebService/src/main/java/com/sintraqos/portfolioprojectAPI/webservice/controllers/WebAuthController.java:
```diff
@@ -0,0 +1,19 @@
+     */
+    @GetMapping("/login")
+    public String getLoginPage(Model model) {
+        return "login";
```
eer zijn geen templates in het project zo ver ik kan zien. wat returned "login" precies?

gglerum commented on PortfolioProjectAPI/PortfolioProjectAPI.WebService/src/main/java/com/sintraqos/portfolioprojectAPI/webservice/WebServiceConfig.java:
```diff
@@ -0,0 +1,13 @@
+    // Define beans specific to this module
+    @Bean
+    public WebServiceConfig portfolioService() {
+        return new WebServiceConfig();
```
hij returned zichzelf? waar wordt dit precies voor gebruikt?

gglerum commented on PortfolioProject/PortfolioProject.Core/src/main/java/com/sintraqos/portfolioproject/CoreBeans.java:
```diff
@@ -0,0 +1,19 @@
+@Configuration
+public class CoreBeans {
+
+    // Display all current loaded beans 
```
deze methode mag weg

gglerum commented on PortfolioProjectAPI/pom.xml:
even nakijken of je wel alle dependencies uit deze pom gebruikt.

Sintraqos committed:
Update

Added a bunch of tests to the app.

gglerum commented on PortfolioProject/pom.xml:
```diff
@@ -0,0 +1,144 @@
+    <packaging>pom</packaging>
+    <name>spring</name>
+	<description>Demo project for Spring Boot</description>
+	<url/>
```
deze lege tags invullen of weglaten.

gglerum commented on PortfolioProject/create.sql:
is het de bedoeling dat dit bestand leeg is en blijft?

gglerum commented on PortfolioProject/PortfolioProject.Core/src/main/java/com/sintraqos/portfolioproject/Main.java:
```diff
@@ -0,0 +1,45 @@
+import org.springframework.scheduling.annotation.EnableScheduling;
+
+@SpringBootApplication(scanBasePackages = "com.sintraqos.portfolioproject")
+@Import({WebServiceConfig.class, Testing.class})  // Import the WebServiceConfig class
```
Zou Testing.class niet gebruikt moeten worden in een Test config of in een class van je testen?

gglerum commented on PortfolioProject/PortfolioProject.Forum/src/test/java/com/sintraqos/portfolioproject/forum/forumPost/service/ForumPostServiceTest.java:
```diff
@@ -0,0 +1,51 @@
+import org.springframework.stereotype.Service;
+
+@Service
+class ForumPostServiceTest {
```
Deze testen werken niet zonder Assertions. Wanneer je ze automatisch draait via maven, dan zullen alle testen altijd slagen, ongeacht of het wel of niet verkeerd gaat.

gglerum commented on PortfolioProject/PortfolioProject.Forum/src/test/java/com/sintraqos/portfolioproject/forum/forumPost/useCase/UseCaseAddForumPostTest.java:
```diff
@@ -0,0 +1,76 @@
+
+        System.out.printf("Added new message: '%s'%n", forumPost.getMessage());
+
+        Assertions.assertTrue(true);
```
????

gglerum commented on PortfolioProject/PortfolioProject.Scheduler/src/main/java/com/sintraqos/portfolioproject/scheduler/Scheduler.java:
```diff
@@ -0,0 +1,23 @@
+import org.springframework.stereotype.Component;
+
+@Component
+public class Scheduler {
```
Wat wordt er precies gescheduled?

gglerum reviewed (changes requested):
Je testen moet je nog even nakijken. Die werken niet met assertions en gaan er vanuit dat je altijd de console in de gaten houd. Wanneer je met assertions werkt, dan stopt de built als de testen niet slagen, op die manier kunnen er ook geen bugs op productie terecht komen.

Sintraqos committed:
Update

GameReviewAPI module now uses a JSON file for creating reviews which are visible on the home page

Sintraqos committed:
Update

Added some comments to code.
Changed the scheduler and the scheduledHandler to a dedicated eventHandler for retrieving games.
Removed the Testing class and all references to it as it is no longer needed

Sintraqos committed:
Update

Redone the tests of the forumPosts and started those inside the games

Sintraqos committed:
Update

Almost finished writing up the tests of the AccountUseCases

Sintraqos committed:
Update

Finished writing the tests for the User, ForumPosts and Library

Sintraqos committed:
Update

Cleaned up both projects form unneeded / unnecessary scripts and/or code

Sintraqos committed:
Update

Removed more unneeded files and code.
Removed unneeded imports.
Moved the GameServiceClient to the API folder, so now all API related code is in 1 single module.
Updated the accountSettings html page. Users can now delete their accounts.
Moved the caching scripts to their own module.
Updated the caching so now it will cache the User and Game repositories too.
Updated the HTML pages so it will now display a Footer, as defined in the Fragments.html and Styles.css

Sintraqos committed:
Update HTML pages

Fixed the header and footer fragments
Added in button states for the button instead of a singular sprite.
Moved the images to their respective folders instead of all in the same folder.
Updated login page to use the same style as used in the settings

Sintraqos committed:
Update

Fixed login and register pages so they now show the proper buttons with alignment
Made sure the messages the server sends to the user now align to the center
Added in a new button to use which spans the entire width of the screen,

Sintraqos committed:
Update

Added in a logo.
The logo displays on the left side of the header and now in the tab inside the browser itself
Removed the <style> from the library page so that the styles.css will handle the style instead of the page itself

Sintraqos committed:
Update

Fixed role setting for users.

Sintraqos committed:
Update

Sintraqos committed:
Update

Back End:
- Added comments to scripts.
- Updated the Usermessage, one of the constructers no longer need both a DTO and an Entity, only one or the other, for improved readability for the contstructor
- Changed the import formatting and order. Merged imports where possible, and made the order the same across all scripts.

Front End:
- Added in a font that the website now uses.
- Added in a label, for text with a background
- Added in a custom selection box
- The adminDashboard now properly shows all the Admin accounts
- Forum posts now get displayed in the new Label object
- The role change now uses the custom selection box

Sintraqos committed:
Update

Added in Forum Post Validation, and removed the validation from the UseCaseAddForumPost  itself, now it only uses the validation from the UseCaseValidateForumPost.
- Checks if the user is able to post based on the time the user previously posted a message
- Checks if the message doesn't contain too many repeating characters, IE: AAAA
- Checks if the message doesn't contain an SQL query, URL or code fragment
- Checks if the message doesn't contain non-supported characters
- Added in the necessary error messages to the Errors class
Removed the caching from ForumPostRepository.findAllByAccountIDOrderByPostDateDesc, to prevent a potential memory leak from a lot of users posting messages.

Sintraqos committed:
Update

Added in a validation for adding games, users or forum posts to their respective repositories

Sintraqos committed:
Update

Fixed the ValidateForumPost
- It now checks at the end to censor the word, this will prevent from false flagging the repeated characters as it did previously

Sintraqos committed:
Update

Censorship now works again for the forumPosts. It will now use the validatedForumPost instead of the base one to post

Sintraqos committed:
Update

Push of final version of project.
No (real) changes will be uploaded from this point forward
